packages/bitcore-lib/lib/transaction/transaction.js

```diff
@@ -632,7 +632,17 @@ Transaction.prototype.from = function(utxo, pubkeys, threshold, opts) {
   return this;
 };
 
-Transaction.prototype._fromNonP2SH = function(utxo) {
+Transaction.prototype.associateInputs = function(utxos) {
+  for(let utxo of utxos) {
+    const index = this.inputs.findIndex(i => i.prevTxId.toString('hex') === utxo.txId && i.outputIndex === utxo.outputIndex);
+    if(index >= 0) {
+      this.inputs[index] = this._getInputFrom(utxo);
+    }
+  }
```

Comment:
It would be nice to return the indexes of the associated UTXOs.

Reply:
okay, I'll add that